File: syno_backend/src/main/java/com/syno_back/backend/dto/UpdateUserTranslationDto.java
package com.syno_back.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import javax.validation.constraints.Size;
import java.time.LocalDateTime;

/**
 * DTO for updating <code>DbTranslation</code>
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Builder
@ToString
public class UpdateUserTranslationDto {
    /**
     * Unique <code>DbTranslation</code> id
     */
    @JsonProperty("pin")
    private String pin;

    /**
     * Actual translation
     */
    @Size(max=100)
    @JsonProperty("translation")
    private String translation;

    /**
     * User's comment
     */
    @Size(max=100)
    @JsonProperty("comment")
    private String comment;

    /**
     * Translation's transcription
     */
    @Size(max=100)
    @JsonProperty("transcription")
    private String transcription;

    /**
     * Usage sample
     */
    @Size(max=100)
    @JsonProperty("usage_sample")
    private String usageSample;

    /**
     * Time when translation was created
     */
    @JsonProperty("time_created")
    private LocalDateTime timeCreated;

    /**
     * Time when translation was modified
     */
    @JsonProperty("time_modified")
    private LocalDateTime timeModified;
}

File: syno-mobile/syno-mobile/Core/Dtos/UpdateUserTranslationDto.swift
import Foundation

/// DTO for updating `DbTranslation` on server
class UpdateUserTranslationDto: Codable {
    let pin: String
    
    let translation: String
    
    let comment: String
    
    let transcription: String
    
    let usageSample: String
    
    let timeCreated: Date?
    
    let timeModified: Date?
    
    /**
     Creates new ``
     - Parameters:
        - pin: `DbTranslation` unique id
        - translation: actual translation
        - comment: user's comment
        - usageSample: user's provided sample
        - timeCreated: time when translation was created
        - timeModified: time when translation was modified
     */
    init(pin: String, translation: String, comment: String, transcription: String, usageSample: String, timeCreated: Date?, timeModified: Date?) {
        self.pin = pin
        self.translation = translation
        self.comment = comment
        self.transcription = transcription
        self.usageSample = usageSample
        self.timeCreated = timeCreated
        self.timeModified = timeModified
    }
}
